src/test/java/it/gov/pagopa/payhub/auth/service/a2a/retrieve/ClientRetrieverServiceTest.java

```diff
@@ -0,0 +1,60 @@
+package it.gov.pagopa.payhub.auth.service.a2a.retrieve;
+
+import it.gov.pagopa.payhub.auth.model.Client;
+import it.gov.pagopa.payhub.auth.repository.ClientRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+@ExtendWith(MockitoExtension.class)
+class ClientRetrieverServiceTest {
+	@Mock
+	private ClientRepository clientRepositoryMock;
+
+	@InjectMocks
+	private ClientRetrieverService service;
+
+	@Test
+	void givenGetClientIdWhenReturnStoredClientSecretThenInvokeClientRetrieverService(){
```

Comment:
in the `when` clause lets use the tested method name

Reply:
in https://github.com/pagopa/p4pa-auth/pull/86